tokio-util/src/sync/reusable_box.rs

```diff
@@ -47,24 +45,14 @@ impl<'a, T> ReusableBoxFuture<'a, T> {
     where
         F: Future<Output = T> + Send + 'a,
     {
-        // If we try to inline the contents of this function, the type checker complains because
-        // the bound `T: 'a` is not satisfied in the call to `pending()`. But by putting it in an
-        // inner function that doesn't have `T` as a generic parameter, we implicitly get the bound
-        // `F::Output: 'a` transitively through `F: 'a`, allowing us to call `pending()`.
-        #[inline(always)]
-        fn real_try_set<'a, F>(
-            this: &mut ReusableBoxFuture<'a, F::Output>,
-            future: F,
-        ) -> Result<(), F>
-        where
-            F: Future + Send + 'a,
+        if size_of_val(&*self.boxed) == size_of_val(&future)
+            && align_of_val(&*self.boxed) == align_of_val(&future)
         {
-            // future::Pending<T> is a ZST so this never allocates.
-            let boxed = mem::replace(&mut this.boxed, Box::pin(future::pending()));
-            reuse_pin_box(boxed, future, |boxed| this.boxed = Pin::from(boxed))
+            self.boxed = Box::pin(future);
```

Comment:
The point of `ReusableBoxFuture` is to reuse the `Box`, but this appears to create a new box?

Reply:
Yeah, sorry, my bad. I'm going to close this